libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,262 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    ArrowRight as ArrowRightIcon,
+    Clear as ClearIcon,
+} from '@mui/icons-material/';
+import { Box, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useEffect, useRef, useState } from 'react';
+import { asFormattedMediaTimecodeString } from './shared/as-formatted-media-timecode-string';
+import { AudioMIMEType } from './shared/audio-mime-type.enum';
+import { isAudioMIMEType } from './shared/is-audio-mime-type';
+import { KeyboardKey, useKeyDown } from './shared/use-key-down';
+import { TimeRangeSelectionVisual, TimeRangeVisualState } from './time-range-selection-visual';
+
+type Nullable<T> = T | null;
+
+export type MedidaPlayDirection = 'forward' | 'reverse';
+
+export enum KeyboardShortcuts {
+    play = KeyboardKey.spacebar,
+    clear = KeyboardKey.c,
+    markInPoint = KeyboardKey.i,
+    markOutPoint = KeyboardKey.o,
+    scrubForward = KeyboardKey.k,
+    scrubBackward = KeyboardKey.j,
+}
+
+const StyledAudioPlayer = styled('audio')`
+    border-radius: 20px;
+`;
+
+export type TimeRangeSelection = {
+    inPointSeconds: number;
+    outPointSeconds: number;
+};
+
+interface AudioAnnotatorProps {
+    audioUrl: string;
+    mimeType?: AudioMIMEType;
+    onTimeRangeSelected: (timeRangeSelected: Nullable<TimeRangeSelection>) => void;
+}
+
+export const AudioAnnotator = ({
+    audioUrl,
+    mimeType,
+    onTimeRangeSelected,
+}: AudioAnnotatorProps) => {
+    const audioRef = useRef<HTMLVideoElement>(null);
+
+    const [inPointSeconds, setInPointSeconds] = useState<Nullable<number>>(null);
```

Comment:
I tried to go with `timeRangeSelection` as the root state but it didn't seem to work well with the JSX template because I was always dealing with nested null checks (first check `timeRangeSelection` is not null, then check `inPointSeconds` is not null).  

So with this scenario, the `timeRangeSelection` is only used to pass the selection back to the parent presenter component.

Also, semantically I think this makes sense because a `timeRangeSelection` with only `inPointSeconds` selected is by definition an invalid time range selection because it lacks an `outPointSeconds`.